addons/udes_stock/controllers/stock_picking.py

```diff
@@ -58,3 +58,32 @@ def is_compatible_package(self, ident, package_name):
         if not picking.exists():
             raise ValidationError(_('Cannot find stock.picking with id %s') % ident)
         return picking.is_compatible_package(package_name)
+
+    @http.route('/api/stock-picking/<identifier>/suggested-locations',
+                type='json', methods=['GET'], auth='user')
+    def suggested_locations(self, identifier, package_name=None,
+                            move_line_ids=None):
+        """ Search suggested locations
+
+            Example output:
+                { "jsonrpc": "2.0",
+                  "result" : [
+                    {"id": 1, "name": "Location 1", "barcode": "L00000100"},
+                    {"id": 2, "name": "Location 2", "barcode": "L00000200"}
+                ]}
+        """
+        Package = request.env['stock.quant.package']
+        MoveLine = request.env['stock.move.line']
+        Picking = request.env['stock.picking']
+
+        picking = Picking.browse(int(identifier))
+
+        kwargs = {}
+        if package_name:
+            package = Package.get_package(package_name)
+            kwargs['package'] = package
+        elif move_line_ids:
+            kwargs['move_line_ids'] = MoveLine.browse(move_line_ids)
```

Comment:
I would raise a `ValidationError` in an `else` right above.

Reply:
No need for that ^^.
After chatting, we agreed that we don't need to validate the request here as we want to keep the format loose, being policy-specific.